### PR TITLE
docs: update marko examples with new apis

### PR DESCRIPTION
**Pull request by DylanPiercey**
# Description
This contains a few modernizations to the Marko testing library docs.

Specifically it references the `screen` api which is now fully supported there.
It also mentions the additional `renderResult.cleanup` api.
it adds a note about component render timing and awaiting the `fireEvent` api.
Finally it updates the jest section with the latest examples.

### CODE REVIEW

File: docs/marko-testing-library/api.mdx [hunk 1]
```diff
@@ -212,6 +215,32 @@ expect(emitted()).toMatchInlineSnapshot(`
   `)
 ```
 
+### `cleanup`
+
+Like the top level cleanup method cleanup method, this allows you to remove and
```
Comment (eps1lon) [inline docs/marko-testing-library/api.mdx:220]:
```suggestion
Like the top level cleanup method, this allows you to remove and
```

**Review by eps1lon — CHANGES_REQUESTED** [review]
Can't review the substance of the PR. Just a minor typo. Once that is fixed, we can merge.

File: docs/marko-testing-library/api.mdx (hunk 1 above)
Comment (DylanPiercey) [reply docs/marko-testing-library/api.mdx]: Good catch, thanks!

I went ahead and put the Marko Testing Library documentation through Grammarly to catch some issues like this and made a bit more cleanup / link fixes.

**Comment by eps1lon** [comment]
@all-contributors add @DylanPiercey for docs

**Pull request merged**